### PR TITLE
[FIX] stock_barcode: set procurement group on new moves

### PR DESCRIPTION
**Pull request by HANNICHE-Walid**
Steps to reproduce:
- Confirm an SO with a storable product
- in barcode app open the created picking
- Scan a different product confirm the picking and create a backorder

Bug:
the new product is not added to the SO and the original picking isn't linked the SO anymore

Fix:
when new moves are created in the inventory app default_picking_id is set in the context and is used in _default_group_id to set the group

set the context key during the write to ensure moves are created with the correct group
test: https://github.com/odoo/enterprise/pull/57408

opw-3644773

### CODE REVIEW

File: addons/stock/models/stock_move.py [hunk 1]
```diff
@@ -576,6 +576,15 @@ def name_get(self):
...
         return res
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if 'picking_id' in vals and 'group_id' not in vals:
+                picking = self.env['stock.picking'].browse(vals['picking_id'])
+                if picking.group_id:
+                    vals['group_id'] = picking.group_id.id
+        return super().create(vals_list)
```
Comment (Whenrow) [inline addons/stock/models/stock_move.py:586]: could you add the same idea in `write` too ? I can think of this 
https://github.com/odoo/odoo/blob/23c805030abb89ca449cf430d5661d725f5801a8/addons/stock/models/stock_move.py#L1126 
for instance

**Comment by HANNICHE-Walid** [comment]
hi @Whenrow I've added the fix to the write aswell

**Comment by Whenrow** [comment]
robodoo r+

**Comment by HANNICHE-Walid** [comment]
@robodoo retry

**Comment by HANNICHE-Walid** [comment]
@Whenrow I fixed the linked enterprise PR

**Comment by Whenrow** [comment]
robodoo retry

**Comment by rousseldenis** [comment]
@HANNICHE-Walid @Whenrow Why wasn't the test put in stock module (as I see test code does not depends on stock_barcode code...) ?

**Comment by HANNICHE-Walid** [comment]
> test code does not depends on stock_barcode

@rousseldenis  it does ...
if you add a product on the picking in the inventory app the bug isn't triggered since the picking_id is set in the context

**Comment by rousseldenis** [comment]
> > test code does not depends on stock_barcode
> 
> @rousseldenis it does ... if you add a product on the picking in the inventory app the bug isn't triggered since the picking_id is set in the context

Ok, nevertheless, I would have created the test that programmatically create the backorder to see the group is transferred from the original picking.

**Comment by HANNICHE-Walid** [comment]
Hi @rousseldenis as I previously said this bug only happens in the context of barcode
the initial fix was completely on the enterprise side by setting the correct key in the context
after further discussion we decided that directly setting the value during write and create is more robust than relying on setting a default value in the context

**Comment by HANNICHE-Walid** [comment]
> I would have created the test here

feel free to do so if there's a use case I'm missing
it's an open-source project :)

**Pull request closed without merging**